httpgrpc/json.go

```diff
@@ -0,0 +1,41 @@
+package httpgrpc
+
+import (
+	"bytes"
+
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/golang/protobuf/proto"
+	"google.golang.org/grpc/encoding"
+)
+
+var (
+	grpcJsonMarshaler = jsonpb.Marshaler{
+		EnumsAsInts:  true,
+		EmitDefaults: true,
+	}
+
+	grpcJsonUnmarshaler = jsonpb.Unmarshaler{
+		AllowUnknownFields: true,
+	}
+)
+
+func init() {
+	encoding.RegisterCodec(jsonCodec{})
```

Comment:
This has a side effect of possibly overwriting the registration of something else that is also interested in JSON marshaling with gRPC. I'm not sure that's what we want.